File: src/main/java/fr/epicanard/mapsaver/models/map/MapByName.java
package fr.epicanard.mapsaver.models.map;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.Optional;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Getter
public class MapByName {
    private UUID mapUuid;
    private UUID playerUuid;
    private Visibility visibility;
    private Optional<Integer> lockedId;
    private Optional<String> server;
}

File: src/main/scala/fr/epicanard/mapsaver/models/map/MapByName.scala
package fr.epicanard.mapsaver.models.map

import java.util.UUID

case class MapByName(
    playerUuid: UUID,
    dataId: Int,
    visibility: Visibility,
    lockedMap: Option[LockedMap],
    mapInfo: McMapInfo
)

case class LockedMap(
    lockedId: Int,
    server: String
)
